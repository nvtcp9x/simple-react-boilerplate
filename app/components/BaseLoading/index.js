// @flow

import * as React from 'react';
import Loading from '../Loading'

type Props = {
  loading: boolean,
  children: React.Element<>
}
const BaseLoading = ({ loading, children }: Props) => (loading ? <Loading /> : children)

export default BaseLoading;
