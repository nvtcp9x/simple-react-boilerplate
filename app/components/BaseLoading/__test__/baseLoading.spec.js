import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import BaseLoading from '../index';
import { toBuffer } from 'ip';

Enzyme.configure({ adapter: new Adapter() });

describe('BaseLoading component', () => {
  it('should match to SnapShot', () => {
    const component = shallow(<BaseLoading />);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should render <Loading />', () => {
    const mockProps = {
      loading: true,
      children: null,
    }
    const component = mount(<BaseLoading {...mockProps} />)
    expect(component.find('div iframe').length).toBe(1);
  })
});
