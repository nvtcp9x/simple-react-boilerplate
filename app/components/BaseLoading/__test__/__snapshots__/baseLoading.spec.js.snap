// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLoading component should match to SnapShot 1`] = `""`;
